dali/base/dasess.cpp

```diff
@@ -453,6 +453,54 @@ class CEnableScopeScansReq : implements IMessageWrapper
     }
 };
 
+
+//Helper class to Serialize/deserialize digital signature of "key;user;timestamp" and request timestamp
+class CSignatureSerializationHelper
+{
+public:
+    void serializeSignature(const char * key, IUserDescriptor *udesc, MemoryBuffer &mb)
+    {
+        if (queryDaliServerVersion().compare("3.15") >= 0)
+        {
+            IDigitalSignatureManager * pDSM = createDigitalSignatureManagerInstanceFromEnv();
+            if (pDSM && pDSM->isDigiSignerConfigured())
+            {
+                //Serialize key, timestamp and signature of "key;username;timestamp"
+                //Dali will use this to ensure request came from
+                //valid authenticated user within a reasonable timeframe
+                StringBuffer username;
+                udesc->getUserName(username);
+                CDateTime now;
+                now.setNow();
+                StringBuffer timeStr;
+                now.getString(timeStr, false);//get UTC timestamp
+                VStringBuffer toSign("%s;%s;%s", key, username.str(), timeStr.str());
+
+                StringBuffer b64sig;
+                pDSM->digiSign(toSign, b64sig);//Sign "username;timeStamp"
+
+                //Serialize the signature, and the timestamp object
+                mb.append(b64sig.str());
+                now.serialize(mb);
+            }
+        }
+    }
+
+    void deserializeSignature(MemoryBuffer &mb, StringBuffer & signature, CDateTime & utcTimeStamp)
+    {
+        if (queryDaliServerVersion().compare("3.15") >= 0)
+        {
+            if (mb.remaining() > 0)
+            {
+                mb.read(signature);
+                if (mb.remaining() > 0)
```

Comment:
don't think this is required. If >= 3.15 - will have neither or both signature + datestamp serialized